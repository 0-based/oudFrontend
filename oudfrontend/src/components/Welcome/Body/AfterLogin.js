import React, {Component} from 'react';
import {Link, Redirect} from 'react-router-dom';
import {CardList} from './../item/cardlist/cardlist';
import adele from './../../../assets/images/adeleImg.png';
import Back from './../../../assets/images/2685063.jpg';
import oud from './../../../assets/images/oud.png';
class AfterLogin extends Component {
  constructor(props) {
    super(props);
    this.state = {
      MusicCard: [
        {
          id: '1',
          picSrc: {adele},
          name: 'Hello',
          actor: 'adele',
        },
        {
          id: '2',

          picSrc: {adele},
          name: 'Hello',
          actor: 'adele',
        },
        {
          id: '3',

          picSrc: {adele},
          name: 'Hello',
          actor: 'adele',
        },
        {
          id: '4',

          picSrc: {adele},
          name: 'Hello',
          actor: 'adele',
        },
        {
          id: '5',

          picSrc: {adele},
          name: 'Hello',
          actor: 'adele',
        },
        {
          id: '6',

          picSrc: {adele},
          name: 'Hello',
          actor: 'adele',
        },
      ],
    };
  }
  render() {
    return (
      <div className="BodyStyle">
        <img
          src={Back}
          data-testid="background"
          className="backGround"
          alt="background"
        />
        <div className="welcomeTextUser">
          <img src={oud} alt="oud" className="oudLogoBody" />
          <h1 data-testid="firstText">Music for everyone.</h1>
          <h6 data-testid="secText">
            Millions of songs. No credit card needed.
          </h6>
          <Link className="LinkStyle" to="/premium">
            <button className="getOudBtn" data-testid="getOudBtn">
              Get Oud Premium
            </button>
          </Link>
        </div>
        <div className="container">
          <div className="bottom">
            <h1>Looking for music?</h1>
            <h5>Start listening to the best new releases.</h5>
            <Link className="LinkStyle" to="/">
              <button data-testid="getOudBtn2" className="getOudBtn2">
                LAUNCH WEB PLAYER
              </button>
            </Link>
          </div>
        </div>
        <div className="container">
          <div className="MusicCard">
            <CardList MusicCard={this.state.MusicCard}></CardList>
          </div>
        </div>
      </div>
    );
  }
}

export default AfterLogin;
