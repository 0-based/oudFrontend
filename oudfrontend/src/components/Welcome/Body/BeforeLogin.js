import React, { Component } from 'react'
import { Link } from 'react-router-dom';
import Back from './../../../assets/images/2685063.jpg';

class BeforeLogin extends Component {
  render() {
    return (
      <div className="BodyStyle">
        <img
          src={Back}
          className="backGround"
          alt="background"
          data-testid="background"
        ></img>
        <div className="welcomeTextGeust">
          <h1 data-testid="firstText">Music for everyone.</h1>
          <h6 data-testid="secText">
            Millions of songs. No credit card needed.
          </h6>
          <Link to="/SignUP">
            <button className="getOudBtn" data-testid="getOudBtn">
              Get Oud Free
            </button>
          </Link>
        </div>
      </div>
    )
  }
}

export default BeforeLogin
