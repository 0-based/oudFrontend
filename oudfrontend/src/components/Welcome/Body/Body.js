import React, { Component } from 'react'

class Body extends Component {
  render() {
    return (
      <div>
        Body is Loaded !!
      </div>
    )
  }
}

export default Body
