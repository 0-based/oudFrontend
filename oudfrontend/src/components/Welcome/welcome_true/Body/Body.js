import React, {Fragment, useState, Component} from 'react';
import {Link, Redirect} from 'react-router-dom';
import Back from '../../../../assets/images/2685063.jpg';
import '../../welcome.css';
/**
 * @class Body
 * the body class
 */
class Body extends Component {
  constructor(props) {
    super(props);
    this.state = {redirect1: false, redirect2: false};
  }
  setRedirect1 = () => {
    this.setState({
      redirect1: true,
    });
  };
  setRedirect2 = () => {
    this.setState({
      redirect2: true,
    });
  };
  /**
   * this is the function that have the path of the page that i want to go
   * @function
   * @returns {void}
   */
  toSignUP = () => {
    if (this.state.redirect1) {
      return <Redirect to="/SignUP"></Redirect>;
    }
  };
  towebplayer = () => {
    if (this.state.redirect2) {
      return <Redirect to="/webPlayer"></Redirect>;
    }
  };
  /**
   * @function
   * @returns {JSX}
   */
  render() {
    return (
      <div className="BodyStyle">
        <img
          src={Back}
          data-testid="background"
          className="backGround"
          alt="background"
        ></img>
        <div className="welcomeTextUser">
          <h1 data-testid="firstText">Music for everyone.</h1>
          <h6 data-testid="secText">
            Millions of songs. No credit card needed.
          </h6>
          {this.redirect1 ? this.toSignUP() : null}
          <Link className="LinkStyle" to="/premium">
            <button className="getOudBtn" data-testid="getOudBtn">
              Get Oud Premium
            </button>
          </Link>
        </div>
        <div className="container">
          <div className="bottom">
            <h1>Looking for music?</h1>
            <h5>Start listening to the best new releases.</h5>
            {this.redirect2 ? this.towebplayer() : null}
            <Link className="LinkStyle" to="webPlayer">
              {' '}
              <button
                data-testid="getOudBtn2"
                className="getOudBtn2"
                onClick={this.setRedirect2}
              >
                LAUNCH WEB PLAYER
              </button>
            </Link>
          </div>
        </div>
      </div>
    );
  }
}

export default Body;
