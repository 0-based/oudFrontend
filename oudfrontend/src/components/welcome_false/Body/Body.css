.backGround {
    width: 100%;
}

.Body {
    position: relative;
    display: inline-block;
    width: 100%;
}


.welcomeText {
    position: absolute;
    margin: 0 auto;
    left: 0;
    right: 0;
    top: 40%;
    text-align: center;
    color: white;
}

@media (max-width: 1100px) {
    .welcomeText {
        top: 40%;
    }

    h1 {
        font-size: 30px;
    }

    h6 {
        font-size: 10px;
    }

    .welcomeText button {
        width: 32%;
        font-size: 100%;
    }
}

@media (max-width: 550px) {
    .welcomeText {
        top: 18%;
    }
}

.getOudBtn {
    background-color: #f9b835;
    width: 15.6rem;
    height: 2.5rem;
    border: none;
    border-radius: 50px;
    font-size: 25px;
    color: white;
    font-family: 'Lalezar',
        cursive;
    align-items: center;
    align-content: center;
}


@media (max-width:800px) {}

.bottom {
    padding-top: 20px;
    position: relative;
    margin: 0 auto;
    left: 0;
    right: 0;
    text-align: center;
    color: black;
    font-family: inherit;
}

.bottom h1 {
    font-size: 50px;
}

.bottom h5 {
    padding-bottom: 5px;
}

.getOudBtn2 {
    width: 15.6rem;
    height: 2.8rem;
    border: 1.5px solid gray;
    border-radius: 50px;
    font-size: 15px;
    color: gray;
    padding-top: 3px;
    font-family: 'Lalezar', cursive;
    margin-bottom: 50px;
    margin-top: 10px;
}

.getOudBtn2:hover {
    border: 1.5px solid gray;
    color: white;
    background-color: gray;
    transition: .5s;
}

.getOudBtn2:active {
    border: 1.8px solid black;
    color: white;
    background-color: black;
    transition: .5s;
}