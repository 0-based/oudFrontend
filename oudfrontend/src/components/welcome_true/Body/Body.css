.backGround {
    width: 100%;
}

.Body {
    position: relative;
    display: inline-block;
    width: 100%;
    /* Make the width of box same as image */
}

/* button {
    color: white;
    position: absolute;
    margin: 0 auto;
    left: 0;
    right: 0;
    top: 55%;
    text-align: center;

} */

.welcomeText {
    position: absolute;
    margin: 0 auto;
    left: 0;
    right: 0;
    top: 40%;
    text-align: center;
    color: white;
}

@media (max-width: 600px) {
    .welcomeText {
        top: 30%;
    }
    h1{
        font-size: 30px;
    }
    h6{
        font-size: 10px;
    }
    .welcomeText button{
        width: 30%;
        font-size: 100%;

    }
}


.getOudBtn {
    background-color: #f9b835;
    width: 15.6rem;
    height: 2.5rem;
    border: none;
    border-radius: 50px;
    font-size: 25px;
    color: white;
    font-family: 'Lalezar',
        cursive;
    align-items: center;
    align-content: center;
}