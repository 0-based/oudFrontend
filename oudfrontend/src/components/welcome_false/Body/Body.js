import React, {Component} from 'react';
// eslint-disable-next-line no-unused-vars
import {Link} from 'react-router-dom';
import Back from '../../../assets/images/2685063.jpg';
import './Body.css';

/**the body elements  */
class Body extends Component {
  constructor(props) {
    super(props);
    this.state = {};
  }
  /**
   * the render function
   * @function
   * @returns {JSX}
   */
  render() {
    return (
      <div className="Body">
        <img
          src={Back}
          className="backGround"
          alt="background"
          data-testid="background"
        ></img>
        <div className="welcomeTextGeust">
          <h1 data-testid="firstText">Music for everyone.</h1>
          <h6 data-testid="secText">
            Millions of songs. No credit card needed.
          </h6>
          <Link to="/SignUP">
            <button className="getOudBtn" data-testid="getOudBtn">
              Get Oud Free
            </button>
          </Link>
        </div>
      </div>
    );
  }
}

export default Body;
