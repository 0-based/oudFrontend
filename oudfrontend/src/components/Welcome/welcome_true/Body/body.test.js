import React from 'react';
import { render } from '@testing-library/react';
import * as renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom/extend-expect';
import Body from './Body';


describe('Body page testing ', () => {

  it('render correctly text component', () => {
    const TextInputComponent = renderer.create(<Body />).toJSON();
    expect(TextInputComponent).toMatchSnapshot();
  });

  it('render a background ', () => {
    const component = shallow(<Body />);
    const logo = component.find(`[data-testid="background"]`);
    expect(logo.length).toBe(1);
  });

  it('render a Text ', () => {
    const component = shallow(<Body />);
    const Text = component.find(`[data-testid="firstText"]`);
    expect(Text.length).toBe(1);
  });
  it('render a Text ', () => {
    const component = shallow(<Body />);
    const Text = component.find(`[data-testid="secText"]`);
    expect(Text.length).toBe(1);
  });

  it('render a  Button ', () => {
    const component = shallow(<Body />);
    const Button = component.find(`[data-testid="getOudBtn"]`);
    expect(Button.length).toBe(1);
  });
  it('render a  Button ', () => {
    const component = shallow(<Body />);
    const Button = component.find(`[data-testid="getOudBtn2"]`);
    expect(Button.length).toBe(1);
  });
});
