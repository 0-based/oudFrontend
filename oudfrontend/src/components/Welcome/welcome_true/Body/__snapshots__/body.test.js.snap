// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Body page testing  render correctly text component 1`] = `
<div
  className="Body"
>
  <img
    alt="background"
    className="backGround"
    data-testid="background"
    src="2685063.jpg"
  />
  <div
    className="welcomeText"
  >
    <h1
      data-testid="firstText"
    >
      Music for everyone.
    </h1>
    <h6
      data-testid="secText"
    >
      Millions of songs. No credit card needed.
    </h6>
    <button
      className="getOudBtn"
      data-testid="getOudBtn"
    >
      Get Oud Free
    </button>
  </div>
  <div
    className="container"
  >
    <div
      className="bottom"
    >
      <h1>
        Looking for music?
      </h1>
      <h5>
        Start listening to the best new releases.
      </h5>
      <button
        className="getOudBtn2"
        data-testid="getOudBtn2"
      >
        LAUNCH WEB PLAYER
      </button>
    </div>
  </div>
</div>
`;
